Formula/macpine.rb

```diff
@@ -38,6 +38,7 @@ class Macpine < Formula
   def install
     system "make", "install", "PREFIX=#{prefix}"
     generate_completions_from_executable(bin/"alpine", "completion", base_name: "alpine")
+    prefix.install_symlink "bin/alpineDaemonLaunchAgent.plist" => "#{plist_name}.plist"
```

Comment:
I don't think this does what you expect it to. Have you tried testing `brew services start macpine` after rebuilding `macpine` with this change?

Reply:
It seems to place `homebrew.mxcl.macpine.plist` into `/opt/homebrew/opt/macpine` at which point `brew services start macpine` looks like it's installing to `~/Library/LaunchAgents` (I even get a notification from MacOS about it). Is there something I'm missing?

What should that line do / is something incorrect?

Thanks in advance for your help @carlocab!

Reply:
It looks to me like this will just create a broken symlink, which `brew` will clean up. Indeed, downloading the just-built bottles and checking if they contain any `plist` files seems to confirm this:
```
❯ gh run download 4774712800 --pattern 'bottles' -R Homebrew/homebrew-core
❯ for f in bottles/*.tar.gz; do tar tf $f | rg -q plist || echo 'no plist'; done
no plist
no plist
no plist
no plist
no plist
no plist
no plist
```

Reply:
Using a `service` block is probably easier, and should be better supported. Something like this might work:
```diff
diff --git a/Formula/macpine.rb b/Formula/macpine.rb
index 1d480c1d9a0..289444ec332 100644
--- a/Formula/macpine.rb
+++ b/Formula/macpine.rb
@@ -40,6 +40,12 @@ class Macpine < Formula
     generate_completions_from_executable(bin/"alpine", "completion", base_name: "alpine")
   end
 
+  service do
+    run macos: [opt_bin/"alpine", "start", "+launchctl-autostart"]
+    environment_variables PATH: std_service_path_env
+  end
+
   test do
     assert_match "NAME OS STATUS SSH PORTS ARCH PID", shell_output("#{bin}/alpine list")
   end
```
See https://docs.brew.sh/Formula-Cookbook#service-block-methods.

Reply:
I'm not sure at all what that command is doing. If my formula compiles its binary to `bin/` and installs it with `install -m 0744 alpine $(PREFIX)`, is there a better way to similarly install my `plist` file so that `brew services` can pick it up?

This is all [_very_ unclearly documented](https://docs.brew.sh/Formula-Cookbook#service-files) and I must have generated the plist file some other way for it to end up in the right place.

Reply:
Understood. I wanted a platform-independent method that would work for non-brew macOS users, which is why I opted to generate a plist based on the installed path of the binary.

Is there any way I can do that, or should I just use the service block for brew and add a separate method for non-brew?

Thank you again!!

Reply:
Let's use the `service` block. It should have essentially the same content as your generated plist, except:
1. It uses `opt_bin`, which avoids a direct `Cellar` reference (which is not good for `launchctl` services, since these references can go stale after `brew upgrade`)
2. It adds `$HOMEBREW_PREFIX/bin` to `PATH` (which is probably important to find, e.g., `qemu`)

If you wanted to use the generated plist in the formula you would probably have to do something like
```ruby
prefix.install "bin/alpineDaemonLaunchAgent.plist" => "#{plist_name}.plist"
```
but given the two considerations above, let's not do that.